### DISCUSSION TITLE
在armv7l的路由器上运行显示Illegal instruction

### DISCUSSION THREAD

**Discussion by html5syt** [AlistGo/alist#5879] [Q&A]
我有一台路由器，架构是armv7l。我从官方的release页面下载了所有带有arm7字眼的文件，并一个个尝试，除了“alist-linux-musleabihf-armv7l.tar.gz”提示Illegal instruction以外其他的均提示./alist: line 1: syntax error: unexpected word (expecting ")")。
这是我在路由器里执行的uname -a命令：
![Screenshot_20240113_130021_com termux](https://github.com/alist-org/alist/assets/53525043/093e6835-d76c-45aa-a667-658bb405068e)
顺便附带一个从路由器中提取的可正常运行的httpd可执行文件：（把.zip删除）
[httpd.zip](https://github.com/alist-org/alist/files/13926889/httpd.zip)
难道是要我自行编译吗？或者这个架构不是正规armv7l架构？

**Comment by sd14** [comment]
应该是在说openwrt路由器吧？

luci-app-alist有一键安装脚本
sh -c "$(curl -ksS https://raw.githubusercontent.com/sbwml/luci-app-alist/master/install.sh)"

也可手动，openwrt路由器查cpu架构
opkg print-architecture
去下载解压出ipk文件上传路由器安装
https://github.com/sbwml/luci-app-alist/releases/latest
